src/models/gamestate.rs

```diff
@@ -1,17 +1,71 @@
-use super::MetadataOwned;
-use crate::flavor::reencode_float;
-use serde::{Deserialize, Deserializer};
 use std::collections::HashMap;
 
+use serde::{Deserialize, Deserializer};
+
+use crate::Ck3Date;
+use crate::flavor::reencode_float;
+
+use super::MetadataOwned;
+
 #[derive(Debug, Deserialize)]
 pub struct Gamestate {
     pub meta_data: MetadataOwned,
     pub living: HashMap<u64, LivingCharacter>,
+    pub provinces: HashMap<u64, Province>,
+    pub traits_lookup: Vec<String>,
+    pub dynasties: Dynasties,
+    pub religion: Religions,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Dynasties {
+    pub dynasty_house: HashMap<u64, DynastyHouse>,
+    pub dynasties: HashMap<u64, Dynasty>
+}
+
+#[derive(Debug, Deserialize)]
+pub struct DynastyHouse {
+    pub name: Option<String>,
+    pub dynasty: u64
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Dynasty {
+    pub key: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Province {
+    pub holding: Holding,
+    pub fort_level: Option<u64>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Holding {
+    pub r#type: Option<String>,
+    pub buildings: Vec<Building>,
```

Comment:
There are some saves that omit `buildings`, and my guess is that paradox often excludes fields that are empty or their default values, so I tend to prefer to use `#[serde(default)]` and not use Option too much unless there is a difference between empty/0 and `None` in game.